### PR TITLE
Jinja queries support

### PR DESCRIPTION
**Pull request by scholtzan**
Reverts https://github.com/mozilla/bigquery-etl/pull/3689

The issues that caused queries to fail are fixed via https://github.com/mozilla/bigquery-etl/pull/3691/commits/bf8c51810f4cdfcb3d36288b041f17e8990ca1b9

Once merged I'll rerun some of the queries in Airflow that failed yesterday, just to make sure it's also running on Airflow

### CODE REVIEW

File: bigquery_etl/cli/query.py [hunk 1]
```diff
@@ -860,7 +861,20 @@ def _run_query(
...
+            query_stream.write(
+                render_template(
+                    query_file.name,
+                    template_folder=str(query_file.parent),
+                    templates_dir="",
+                    format=False,
+                )
+            )
+
+        with open(tmp.name) as query_stream:
```
Comment (ANich) [inline bigquery_etl/cli/query.py:877]:
```suggestion
            query_stream.seek(0)
```

**Comment by kik-kik** [comment]
Are we able to run this in some kind of dev environment prior to merging to make sure we don't have mass failure again tomorrow morning?

File: bigquery_etl/cli/query.py [hunk 2]
```diff
@@ -860,7 +861,20 @@ def _run_query(
             # point to a public table it needs to be passed as parameter for the query
             query_arguments.append("--destination_table={}".format(destination_table))
 
-        with open(query_file) as query_stream:
+        # write rendered query to a temporary file;
+        # query string cannot be passed directly to bq as SQL comments will be interpreted as CLI arguments
+        tmp = tempfile.NamedTemporaryFile()
+        with open(tmp.name, "w") as query_stream:
```
Comment (ANich) [inline bigquery_etl/cli/query.py:867]:
```suggestion
        with tempfile.NamedTemporaryFile(prefix=query_file.name, mode="w+") as query_stream:
```

**Comment by scholtzan** [comment]
We do run some queries through the `integration` test CI step, however it wasn't running any queries with comments... We can't run queries directly since CI doesn't have access to shared-prod

File: tests/test_run_query.py [hunk 3]
```diff
@@ -34,9 +34,12 @@ def test_run_query(self, tmp_path):
...
 
             assert mock_call.call_args.args == (
-                ["bq", "--dataset_id=test", "--destination_table=query_v1"],
+                [
+                    "bq",
+                    "--dataset_id=test",
+                    "--destination_table=query_v1",
+                ],
             )
-            assert "stdin" in mock_call.call_args.kwargs
```
Comment (ANich) [inline tests/test_run_query.py:39]: We're back to using `stdin` should we re-add these?

File: bigquery_etl/cli/query.py [hunk 4]
```diff
@@ -860,7 +861,20 @@ def _run_query(
             # point to a public table it needs to be passed as parameter for the query
             query_arguments.append("--destination_table={}".format(destination_table))
 
-        with open(query_file) as query_stream:
+        # write rendered query to a temporary file;
+        # query string cannot be passed directly to bq as SQL comments will be interpreted as CLI arguments
+        tmp = tempfile.NamedTemporaryFile()
+        with tempfile.NamedTemporaryFile(prefix=tmp.name, mode="w+") as query_stream:
```
Comment (ANich) [inline bigquery_etl/cli/query.py:867]: Do we still need `tmp` here? We could either omit the `prefix` or make it something meaningful for potential debugging

**Review by ANich — APPROVED** [review]
Left a few comments/suggestions but overall lgtm. It feels like there's some overlap in functionality now between SQL generators and "regular" queries though. This might help us remove some simpler generators

File: sql/mozfun/hist/string_to_json/udf.sql [hunk 5]
```diff
@@ -1,3 +1,4 @@
+{% raw %}
```
[the remark this answers is not in the document]
Comment (scholtzan) [reply sql/mozfun/hist/string_to_json/udf.sql:1]: It has some tests that have `{%s}` in some strings. Jinja is trying to interpret those and is failing

File: bigquery_etl/cli/query.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (sean-rose) [reply bigquery_etl/cli/query.py]: There's still a question about text vs binary mode, but that's probably unlikely to bite us, as the default locale encoding should pretty much always be UTF-8.

**Pull request merged**